suite-native/device/src/hooks/useAuthorizeDevice.ts

```diff
@@ -0,0 +1,84 @@
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { useNavigation, CommonActions } from '@react-navigation/native';
+
+import {
+    selectDeviceRequestedPin,
+    selectIsDeviceConnectedAndAuthorized,
+    selectIsDeviceUnlocked,
+    selectIsNoPhysicalDeviceConnected,
+} from '@suite-common/wallet-core';
+import {
+    RootStackRoutes,
+    ConnectDeviceStackRoutes,
+    HomeStackRoutes,
+} from '@suite-native/navigation';
+
+import { selectIsDeviceReadyToUse } from '../selectors';
+
+const LOADING_TIMEOUT = 2500;
+
+export const useAuthorizeDevice = () => {
+    const [isTimeoutFinished, setIsTimeoutFinished] = useState(false);
+    const navigation = useNavigation();
+
+    const hasDeviceRequestedPin = useSelector(selectDeviceRequestedPin);
+    const isDeviceUnlocked = useSelector(selectIsDeviceUnlocked);
+    const isDeviceReadyToUse = useSelector(selectIsDeviceReadyToUse);
+    const isDeviceConnectedAndAuthorized = useSelector(selectIsDeviceConnectedAndAuthorized);
+    const isNoPhysicalDeviceConnected = useSelector(selectIsNoPhysicalDeviceConnected);
+
+    // The connecting screen should be visible for at least 2.5 seconds before redirecting to HomeScreen.
+    useEffect(() => {
+        const timerId = setTimeout(() => {
+            setIsTimeoutFinished(true);
+        }, LOADING_TIMEOUT);
+
+        return () => clearTimeout(timerId);
```

Comment:
What's the benefit of having this separated from the navigation call compared to how it was before? The conditions present in the other useEffect didn't pass so it didn't get triggered? Because I guess you could wrap the whole navigation link and according to design + technically, it's not needed to display loading screen before pin matrix I think.